numpy/core/function_base.py

```diff
@@ -72,7 +72,9 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
     samples : ndarray
         There are `num` equally spaced samples in the closed interval
         ``[start, stop]`` or the half-open interval ``[start, stop)``
-        (depending on whether `endpoint` is True or False).
+        (depending on whether `endpoint` is True or False).  If start
+        or stop are array-like, then the samples will be along a new
+        axis inserted at the beginning.
```

Comment:
Hmm, wouldn't it make more sense to append the new axis? Is this choice motivated by the intended use? it makes is easier to access all the points by index, OTOH, it is more difficult to access by start-stop pair.

Reply:
This maybe is an implicit request for an `axis` argument. My logic for prepending is that the result will broadcast with `start` and `stop` (and it was easier to code...)